### PR TITLE
fix: bug on request methods

### PR DESCRIPTION
**Pull request by rkdud007**
before, when we send transaction via `request*` method, it checks if the status is not `NONE`, and only if it's none, it scheduled the task and initialize result as "". 

However, as after finish authentication, as the taskStatus turn into `FINALIZED` , which also consider as not `NONE`, if that case, even the task had already stored in mapping, we ignore it and set as `SCHEDULED` and even we overwrite the result. 

This change is intended to fix this issue so that we can route whether if the task had been already finalized, we can emit event about processed result.

### CODE REVIEW

File: src/HdpExecutionStore.sol [hunk 1]
```diff
@@ -122,15 +125,19 @@ contract HdpExecutionStore is Initializable, UUPSUpgradeable, OwnableUpgradeable
         bytes32 datalakeCommitment = blockSampledDatalake.commit();
         bytes32 taskCommitment = computationalTask.commit(datalakeCommitment);
 
-        // Ensure task is not already scheduled
-        if (cachedTasksResult[taskCommitment].status != TaskStatus.NONE) {
-            revert DoubleRegistration();
-        }
+        if (cachedTasksResult[taskCommitment].status == TaskStatus.FINALIZED) {
+            emit TaskAlreadyStored(taskCommitment);
```
Comment (marcellobardus) [inline src/HdpExecutionStore.sol:129]: Shouldn't you return to end the execution?

Comment (rkdud007) [reply src/HdpExecutionStore.sol:129]: But if requested task does not exist, we cannot return the value .Is it possible to choose the return type per branch in solidity?

Comment (rkdud007) [reply src/HdpExecutionStore.sol:129]: Smth like if return type can be Option type this can be solved -- not sure if it doable in sol

Comment (marcellobardus) [reply src/HdpExecutionStore.sol:129]: nvm, all good. Approving

**Pull request merged**